pkg/assembler/graphql/schema/hasSLSA.graphql

```diff
@@ -15,85 +15,114 @@
-  builtFrom: [PkgSrcArtObject!]!
+  "The subject of SLSA attestation: package, source, or artifact."
+  subject: PackageSourceOrArtifact!
+  "The SLSA attestation."
+  slsa: SLSA
+}
+
+"""
+SLSA contains all of the fields present in a SLSA attestation.
+
+The materials and builders are objects of the HasSLSA predicate, everything
+else are properties extracted from the attestation.
+
+We also include fields to specify under what conditions the check was performed
+(time of scan, version of scanners, etc.) as well as how this information got
+included into GUAC (origin document and the collector for that document).
+"""
+type SLSA {
+  "Sources of the build resulting in subject (materials)"
+  builtFrom: [PackageSourceOrArtifact!]!
+  "Builder performing the build"
   builtBy: Builder!
+  "Type of the builder"
   buildType: String!
+  "Individual predicates found in the attestation"
   slsaPredicate: [SLSAPredicate!]!
+  "Version of the SLSA predicate"
   slsaVersion: String!
+  "Timestamp (RFC3339Nano format) of build start time"
   startedOn: Time!
+  "Timestamp (RFC3339Nano format) of build end time"
   finishedOn: Time!
+  "Document from which this attestation is generated from"
   origin: String!
+  "GUAC collector for the document"
   collector: String!
 }
 
 """
 SLSAPredicate are the values from the SLSA predicate in key-value pair form.
-// Predicate:
-"predicateType": "https://slsa.dev/provenance/v1",
+
+For example, given the following predicate
+
+```
 "predicate": {
-    "buildDefinition": {
-        "buildType": string,
-        "externalParameters": object,
-        "systemParameters": object,
-        "resolvedDependencies": [ ...#ArtifactReference ],
+  "buildDefinition": {
+    "externalParameters": {
+      "repository": "https://github.com/octocat/hello-world",
+      ...
     },
-    "runDetails": {
-        "builder": {
-            "id": string,
-            "version": string,
-            "builderDependencies": [ ...#ArtifactReference ],
-        },
-        "metadata": {
-            "invocationId": string,
-            "startedOn": #Timestamp,
-            "finishedOn": #Timestamp,
-        },
-        "byproducts": [ ...#ArtifactReference ],
-    }
+    ...
+  },
+  ...
 }
-where
-"externalParameters": {
-    "repository": "https://github.com/octocat/hello-world",
-    "ref": "refs/heads/main"
-},
+```
+
+we have
+
+```
+key   = "buildDefinition.externalParameters.repository"
+value = "https://github.com/octocat/hello-world"
+```
 
-For example: key = "buildDefinition.externalParameters.repository" value = "https://github.com/octocat/hello-world"
 This node cannot be directly referred by other parts of GUAC.
+
+TODO(mihaimaruseac): Can we define these directly?
 """
 type SLSAPredicate {
   key: String!
   value: String!
 }
 
-"""
-HasSLSASpec allows filtering the list of HasSLSA to return.
-"""
+"HasSLSASpec allows filtering the list of HasSLSA to return."
 input HasSLSASpec {
   package: PkgSpec
```

Comment:
can this all be replaced by `PackageSourceOrArtifactSpec`

Reply:
Right, but that would require more changes on the query side, so I was punting on it. However, I already changed a lot, so will do